### PR TITLE
fix(11576): fixing bug in standardqueue put message

### LINKED ISSUE #11576: bug: Standard SQS queue accepts empty MessageGroupId
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When sending a message to a standard SQS queue in LocalStack, passing empty string as the value for MessageGroupId validates successfully.

### Expected Behavior

When sending a message to a standard SQS queue in LocalStack, passing empty string as the value for MessageGroupId should cause an InvalidParameterValue exception.

### How are you starting LocalStack?

Custom (please describe below)

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker run localstack/localstack

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

```bash
aws sqs send-message \
  --endpoint-url http://localhost:4566 \
  --queue-url "test-queue" \
  --message-body '{}' \
  --message-group-id "" \
  --region us-east-1
```

#### running against LocalStack returns:
```json
{
    "MD5OfMessageBody": "99914b932bd37a50b983c5e7c90ae93b",
    "MessageId": "f4c27de0-bb7a-483b-813f-bed7171d6ac3"
}
```

#### running against AWS returns:
```bash
An error occurred (InvalidParameterValue) when calling the SendMessage operation: Value  for parameter MessageGroupId is invalid. Reason: The request include parameter that is not valid for this queue type.
```

### Environment

```markdown
- OS: macOS Sonoma 14.5
- LocalStack:
  LocalStack version: 3.7.3.dev42
  LocalStack Docker image sha: ed04d31834e012f3e30cd9887f0d2703f01e55399b34e65d01dc2ef701ce1356
  LocalStack build date: 2024-09-24
  LocalStack build git hash: fcd17d9f
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by kshitijkohli**
## Motivation

Raising a fix for an open issue in SQS flow of Localstack.
Fixes #11576 

## Changes

There was a bug in Localstack SQS put message in Standard SQS flow. Principally, standard SQS put calls do not support messagegroupid parameter to be passed. Current implementation of localstack allows "" ie empty string to be passed as messagegroupid while publishing to localstack, whereas same api request to actual AWS sqs throws an exception. Added an additional check in code which checks messagegroupid should not be any string ( having values or empty ) in case of standard sqs put message. If the parameter is an instance of the string, throw exception

### CODE REVIEW

**Comment by kshitijkohli** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by kshitijkohli** [comment]
Hi @thrau , @baermat , @gregfurman  Could you please add the `semver: patch` label to this PR? It contains a small, backward-compatible bug fix. I don't see the access to be able to add the labels. Thank you!

**Review by thrau — CHANGES_REQUESTED** [review]
hey @kshitijkohli, thanks for the contribution!

would you please add a snapshot test case so we can avoid regressions? you can find out more in our contributor docs [here](https://github.com/localstack/localstack/tree/master/docs/testing/parity-testing).

something like this:
```python
    @markers.aws.unknown
    def test_send_message_to_standard_queue_with_empty_message_group_id(self, sqs_create_queue, aws_client, snapshot):
        queue = sqs_create_queue()

        with pytest.raises(ClientError) as e:
            aws_client.sqs.send_message(QueueUrl=queue, MessageBody="message", MessageGroupId="")

        snapshot.match("error", e.value.response)
```

i think if we add and run this we may find that the error message isn't quite correct either. once we understand the full scope of the change i'm happy to set the semver label!

**Comment by kshitijkohli** [comment]
@thrau Pushed the snapshot tests

**Comment by gregfurman** [comment]
Hey @kshitijkohli 😄 

When you ran the snapshot test, did you remember to point your client to AWS? The relevant snippet of the contributing docs that Thomas linked is the [How to write Parity tests](https://github.com/localstack/localstack/tree/master/docs/testing/parity-testing#how-to-write-parity-tests):
> Run the test against AWS: use the parameter `--snapshot-update` (or the environment variable `SNAPSHOT_UPDATE=1`) and set the environment variable as `TEST_TARGET=AWS_CLOUD`.

In addition, this section on [Integration Testing (against AWS)](https://github.com/localstack/localstack/blob/master/docs/testing/integration-tests/README.md) is definitely also relevant!

Once you've setup your testing suite to be able to snapshot against AWS, you can run the test again with those flags enabled:
```shell
make TEST_TARGET=AWS_CLOUD SNAPSHOT_UPDATE=1 TEST_PATH="tests/aws/services/sqs/test_sqs.py::TestSqsProvider::test_send_message_to_standard_queue_with_empty_message_group_id"
```

File: tests/aws/services/sqs/test_sqs.py
Comment (gregfurman) [inline tests/aws/services/sqs/test_sqs.py:442]: Based on the snapshot generated in `tests/aws/services/sqs/test_sqs.snapshot.json`, it doesn't appear inecessary to skip this verification as no `Error.Detail` field exists. 

If you're curious, there's a section in the contributing docs on [skipping verification of snapshot tests](https://github.com/localstack/localstack/blob/master/docs/testing/parity-testing/README.md#skipping-verification-of-snapshot-test) 💡

Comment (kshitijkohli) [reply tests/aws/services/sqs/test_sqs.py]: @gregfurman Yes i ran the code against an sqs i created in my personal aws account and matched the response with the error message that i've updated in my code, but if necessary feel free to generate a snapshot and test on your end as well since it's my first contribution to the repo so still getting caught up with all the documentation :)

Comment (kshitijkohli) [reply tests/aws/services/sqs/test_sqs.py]: @gregfurman Were you able to check the above? 😄

Comment (gregfurman) [reply tests/aws/services/sqs/test_sqs.py]: Hey! Thanks for the follow-up. I just noticed there was no update to the `tests/aws/services/sqs/test_sqs.validation.json`. Did you notice this file update on your local working-tree when you generated your snapshot against AWS?

Otherwise, I tested your changes against AWS and all looks good 😄

Comment (kshitijkohli) [reply tests/aws/services/sqs/test_sqs.py]: No i don't recall any changes happening to this file. Perhaps the way i generated the snapshot may be incorrect, I'm still catching up. From what i understand this file should automatically get an entry with the time on which i generated the snapshot? Let me know if you want me to manually push an entry into this file. Like you said, its working as expected against AWS 😄

Comment (gregfurman) [reply tests/aws/services/sqs/test_sqs.py]: Ahh no worries. I'm happy to just push it from my side if you're happy with that?

Comment (kshitijkohli) [reply tests/aws/services/sqs/test_sqs.py]: Yep works! 😄

**Review by gregfurman — APPROVED** [review]
LGTM 💯  Once the CI passes happy for this to go in. Thanks again for the contribution!

FYI: I just ran the test (`SNAPSHOT_UPDATE=0; TEST_TARGET=AWS_CLOUD`) against AWS to generate that `*.validation.json` 🙂

File: tests/aws/services/sqs/test_sqs.py
Comment (kshitijkohli) [reply tests/aws/services/sqs/test_sqs.py]: @gregfurman I totally forgot about this PR. Is it pending some changes/fixes before it can be merged?

Comment (gregfurman) [reply tests/aws/services/sqs/test_sqs.py]: Ahh thanks for the ping. Should be good to merge 😄 Had to fix a failing test that were fixed after rebase.

**Pull request merged**
